integration_tests/src/main/python/repart_test.py

```diff
@@ -21,8 +21,38 @@
 import pyspark.sql.functions as f
 
 nested_scalar_mark=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/1459")
+# 4 level nested struct
+# each level has a different number of children to avoid a bug in spark < 3.1
+nested_struct = StructGen([
+    ['child0', StructGen([
+        ['child0', StructGen([
+            ['child0', StructGen([
+                ['child0', StructGen([
+                    ['child0', DecimalGen(7, 2)],
+                    ['child1', BooleanGen()],
+                    ['child2', BooleanGen()],
+                    ['child3', BooleanGen()]
+                ])],
+                ['child1', BooleanGen()],
+                ['child2', BooleanGen()]
+            ])],
+            ['child1', BooleanGen()]
+        ])],
+        ['child1', BooleanGen()]
+    ])]])
```

Comment:
It looks like there's still the same number of children at two of the levels?  Maybe I'm misreading the indentation, but it looks like this to me:
```
child0
|-child0
| |-child0
| | |-child0
| | | |-child0
| | | |-child1
| | | |-child2
| | | |-child3
| | |-child1
| | |-child2
| |-child1
|-child1
```
Note how the first and second levels both have 2 children.  Did you mean to drop the last child so the first level only had one child instead of two?

Reply:
Good catch. I have removed one level